manager.py

```diff
@@ -328,40 +343,15 @@ def data_log(self, file, cpm, cpm_err):
                 self.vprint(2, 'Writing CPM to data log at {}'.format(file))
 
     def handle_cpm(self, this_start, this_end):
-        """Get CPM from sensor, display text, send to server."""
-
+        """
+        Get CPM from sensor, display text, send to server.
+        """
         cpm, cpm_err = self.sensor.get_cpm(this_start, this_end)
         counts = int(round(cpm * self.interval / 60))
-
-        start_text = datetime_from_epoch(this_start).strftime(strf)
-        end_text = datetime_from_epoch(this_end).strftime(strf)
-
-        self.vprint(1, CPM_DISPLAY_TEXT.format(
-            time=datetime_from_epoch(time.time()),
-            counts=counts,
-            cpm=cpm,
-            cpm_err=cpm_err,
-            start_time=start_text,
-            end_time=end_text,
-        ))
-        if self.test:
-            self.vprint(
-                1, ANSI_RED + " * Test mode, not sending to server * " +
-                ANSI_RESET)
-            self.data_log(self.datalog, cpm, cpm_err)
-        elif not self.config:
-            self.vprint(1, "Missing config file, not sending to server")
-            self.data_log(self.datalog, cpm, cpm_err)
-        elif not self.publickey:
-            self.vprint(1, "Missing public key, not sending to server")
-            self.data_log(self.datalog, cpm, cpm_err)
-        elif not self.network_up:
-            self.vprint(1, "Network down, not sending to server")
-            self.data_log(self.datalog, cpm, cpm_err)
-        else:
-            self.sender.send_cpm(cpm, cpm_err)
-            self.data_log(self.datalog, cpm, cpm_err)
-
+        self.data_handler.main(self.datalog, cpm, cpm_err, this_start, this_end, counts)
+        if self.quit_after_interval:
```

Comment:
Let's put this in `manager.run` immediately after the `handle_cpm` call. It's the same behavior but I think logically it fits better there (since the quit is not about handling CPM).